scripts/cmake/vcpkg_download_distfile.cmake

```
@@ -123,65 +128,60 @@ function(vcpkg_download_distfile VAR)
-                -l download-${vcpkg_download_distfile_FILENAME}-detailed.log
-                ${request_headers}
-                OUTPUT_FILE download-${vcpkg_download_distfile_FILENAME}-out.log
-                ERROR_FILE download-${vcpkg_download_distfile_FILENAME}-err.log
-                RESULT_VARIABLE error_code
-                WORKING_DIRECTORY ${DOWNLOADS}
-            )
-            if (NOT "${error_code}" STREQUAL "0")
-                message(STATUS
-                    "Downloading ${vcpkg_download_distfile_FILENAME}... Failed.\n"
-                    "    Exit Code: ${error_code}\n"
-                    "    See logs for more information:\n"
-                    "        ${DOWNLOADS}/download-${vcpkg_download_distfile_FILENAME}-out.log\n"
-                    "        ${DOWNLOADS}/download-${vcpkg_download_distfile_FILENAME}-err.log\n"
-                    "        ${DOWNLOADS}/download-${vcpkg_download_distfile_FILENAME}-detailed.log\n"
-                )
-                set(download_success 0)
-            else()
-                file(REMOVE
-                    ${DOWNLOADS}/download-${vcpkg_download_distfile_FILENAME}-out.log
-                    ${DOWNLOADS}/download-${vcpkg_download_distfile_FILENAME}-err.log
-                    ${DOWNLOADS}/download-${vcpkg_download_distfile_FILENAME}-detailed.log
-                )
-                set(download_success 1)
-            endif()
-        else()
             foreach(url IN LISTS vcpkg_download_distfile_URLS)
                 message(STATUS "Downloading ${url} -> ${vcpkg_download_distfile_FILENAME}...")
-                if(vcpkg_download_distfile_HEADERS)
-                    foreach(header ${vcpkg_download_distfile_HEADERS})
-                        list(APPEND request_headers HTTPHEADER ${header})
-                    endforeach()
-                endif()
                 file(DOWNLOAD "${url}" "${download_file_path_part}" STATUS download_status ${request_headers})
                 list(GET download_status 0 status_code)
                 if (NOT "${status_code}" STREQUAL "0")
                     message(STATUS "Downloading ${url}... Failed. Status: ${download_status}")
-                    set(download_success 0)
                 else()
+                    test_hash("${download_file_path_part}" "downloaded file" "The file may have been corrupted in transit. This can be caused by proxies. If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment variables to \"https://user:password@your-proxy-ip-address:port/\".\n")
+                    get_filename_component(downloaded_file_dir "${downloaded_file_path}" DIRECTORY)
+                    file(MAKE_DIRECTORY "${downloaded_file_dir}")
+                    file(RENAME ${download_file_path_part} ${downloaded_file_path})
                     set(download_success 1)
                     break()
                 endif()
             endforeach(url)
+        else()
+            set(URLS)
```

```suggestion
            set(urls)
```

follow naming conventions